### ISSUE TITLE
Notification abonnement on catalogus

### ISSUE THREAD

**Issue by JanBrek** [open-zaak/open-notificaties#156]
Coming forth from a comment in ticket https://github.com/open-zaak/open-zaak/issues/1661 

Allow notification API to be set per catalogus. So set your abbonement to a catalogus instead of using zaaktype as filtering.

One possible solution would be that the zaak kanaal also will include catalogi name/number/whatever in the notification so that it can be used as a filter.

**Comment by joeribekker** [comment]
Refinement: So, Zaken will be filtered in the subscription on the catalog. This is an experimental feature!

Estimate: 5 days

**Comment by joeribekker** [comment]
@JanBrek ping

**Comment by PeterVanBragt** [comment]
Asked Peter Arkesteijn if he has a budget for this

**Comment by PeterVanBragt** [comment]
@joeribekker I can hereby approve this ticket.

**Comment by joeribekker** [comment]
Refinement:

There's 2 options to implement this:

1. We add catalog as a "kenmerk" to the subscription AND the notifications that are sent out.
2. We allow subscribing on catalog and based on the zaaktype-kenmerk, we retrieve the appropriate catalog to decide if a notification should be sent.

We go for option 1 (far less spaghetti) and will retrieve the catalog over the network for external zaaktypes.

**Comment by PeterVanBragt** [comment]
Why is this issue marked as blocked? 🫣

**Comment by stevenbal** [comment]
@PeterVanBragt I forgot to mention it here (https://github.com/open-zaak/open-zaak/pull/1757#issuecomment-2390744492), this change relies on a library which we need to upgrade, but there is another pull request in progress which must be finished first to make use of that upgrade that library in Open Zaak https://github.com/open-zaak/open-zaak/pull/1753

**Comment by joeribekker** [comment]
Refinement: Sonny will continue with the library this week.
